src/Execution/DataLoader/RelationBatchLoader.php

```diff
@@ -77,17 +78,52 @@ function (Model $model): array {
 
     /**
      * Get the parents from the keys that are present on the BatchLoader.
+     *
+     * @return EloquentCollection<mixed>
      */
-    protected function getParentModels(): EloquentCollection
+    protected function getParentModels(bool $shouldLoadRelation = true): EloquentCollection
```

Comment:
This is pretty much duplicated from the `RelationCountBatchLoader`, right? Let's dedupe so we don't miss bugfixes when changing either one of them.

This could be a `public static function` on a utility class, right?

Reply:
Updated. Should clean things up. Thanks for the suggestion.